### PR TITLE
[RSDK-7193] Better state machine and timings for wifi provisioning

### PR DESCRIPTION
**Pull request by Otterverse**
This cuts down on wait times when entering bad (as well as good) passwords during provisioning. Additionally, includes connection errors (e.g. "Bad Password") in device status response and portal page, so reconnecting to a hotspot after a failed attempt will show properly.

Uses new, personal fork of gonetworkmanager, as upstream hasn't responded to (other people's) PRs in over 10 months.

### CODE REVIEW

File: networkmanager/networkmanager.go [hunk 1]
```diff
@@ -587,6 +587,9 @@ func (w *NMWrapper) StartProvisioning(ctx context.Context) error {
 					continue
 				}
 			}
+
+			// signal that the user sent stuff so we can break the main loop
+			userInputChan <- struct{}{}
```
Comment (ale7714) [inline networkmanager/networkmanager.go:592]: 👍

File: networkmanager/networkmanager.go [hunk 2]
```diff
@@ -651,7 +654,9 @@ func (w *NMWrapper) activateConnection(ctx context.Context, ssid string) error {
 	}
 
 	nw.lastTried = now
-	w.lastSSID = ssid
+	if ssid != w.hotspotSSID {
```
Comment (ale7714) [inline networkmanager/networkmanager.go:657]: not sure why this is necessary

File: networkmanager/networkmanager.go [hunk 3]
```diff
@@ -703,21 +708,37 @@ func (w *NMWrapper) deactivateConnection(ssid string) error {
...
-			return errors.Join(err, ErrCouldNotActivateConnection)
+		select {
+		case update := <-changeChan:
+			w.logger.Debugf("%s->%s (%s)", update.OldState, update.NewState, update.Reason)
+			//nolint:exhaustive
+			switch update.NewState {
+			case gnm.NmDeviceStateActivated:
+				return nil
+			case gnm.NmDeviceStateFailed:
+				if update.Reason == gnm.NmDeviceStateReasonNoSecrets {
```
Comment (ale7714) [inline networkmanager/networkmanager.go:732]: 👍

**Review by ale7714 — APPROVED** [review]

File: networkmanager/networkmanager.go (hunk 2 above)
Comment (Otterverse) [reply networkmanager/networkmanager.go:657]: We want to only record the last actual wifi connection attempted, and thus get any errors that happened with it. Without this, the hotspot itself (when it starts up) would always be the "lastSSID" and thus we couldn't fetch errors for the actual network tried.

**Pull request merged**